app/filtering.js

```diff
@@ -372,7 +367,7 @@ function registerPermissionHandler (session, partition) {
       cb(result)
       if (persist) {
         // remember site setting for this host over http(s)
-        appActions.changeSiteSetting(urlOrigin, permission + 'Permission', result, isPrivate)
+        appActions.changeSiteSetting(getOrigin(url), permission + 'Permission', result, isPrivate)
```

Comment:
but shouldn't the permission be set for the page the user visited? Using the `origin` will set the permission for the requesting url